tests/components/onewire/test_options_flow.py

```diff
@@ -0,0 +1,237 @@
+    ):
+        result = await hass.config_entries.options.async_init(config_entry.entry_id)
+    assert result["data_schema"].schema["device_selection"].options == {
+        "Given Name (28.111111111111)": False,
+        "Given Name (28.222222222222)": False,
+        "Given Name (28.222222222223)": False,
+    }
+
+    # Verify that selecting two devices to configure comes back as a
+    #  form with the first device to configure using it's long name as entry
+    with patch(
+        "homeassistant.helpers.device_registry.DeviceRegistry.async_get_device",
+        return_value=FakeDevice(),
+    ):
+        result = await hass.config_entries.options.async_configure(
+            result["flow_id"],
+            user_input={
+                INPUT_ENTRY_DEVICE_SELECTION: [
+                    "Given Name (28.111111111111)",
+                    "Given Name (28.222222222222)",
+                ]
+            },
+        )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert (
+        result["description_placeholders"]["sensor_id"]
+        == "Given Name (28.222222222222)"
+    )
+
+    # Verify that next sensor is coming up for configuration after the first
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={"precision": "temperature"},
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert (
+        result["description_placeholders"]["sensor_id"]
+        == "Given Name (28.111111111111)"
+    )
+
+    # Verify that the setting for the device comes back as default when no input is given
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={"precision": "temperature9"},
+    )
+    assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert (
+        result["data"]["device_options"]["28.222222222222"]["precision"]
+        == "temperature"
+    )
+    assert (
+        result["data"]["device_options"]["28.111111111111"]["precision"]
+        == "temperature9"
+    )
+
+
+async def test_user_owserver_options_no_devices(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    owproxy: MagicMock,
+):
+    """Test that options does not change when no devices are available."""
+    # Initialize onewire hub
+    await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    # Verify that first config step comes back with an empty list of possible devices to choose from
+    result = await hass.config_entries.options.async_init(config_entry.entry_id)
+    await hass.async_block_till_done()
+    assert result["type"] == RESULT_TYPE_ABORT
+    assert result["reason"] == "No configurable devices found."
+
+
+async def test_user_sysbus_options(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+):
+    """Test that SysBus options flow aborts on init."""
+    hass.data[DOMAIN] = {config_entry.entry_id: FakeOWHubSysBus()}
```

Comment:
Don't interact with integration details in the tests. Set up the integration instead with a `MockConfigEntry`, while patching appropriately.